packages/trader/src/AppV2/Hooks/useSignUpHandler.ts

```diff
@@ -0,0 +1,44 @@
+import { useEffect, useState } from 'react';
+import { redirectToSignUp, mobileOSDetectAsync, isSafari } from '@deriv/shared';
+import { useGrowthbookGetFeatureValue } from '@deriv/hooks';
+
+export const useSignupHandler = () => {
+    const [redirect_to_os_signup, setRedirectToOSSignup] = useState(false);
+    const [trigger_os_signup, isGBLoaded] = useGrowthbookGetFeatureValue({
+        featureFlag: 'trigger_os_signup',
+        defaultValue: false,
+    });
+
+    useEffect(() => {
+        if (isGBLoaded) {
+            setRedirectToOSSignup(trigger_os_signup);
+        }
+    }, [isGBLoaded, trigger_os_signup]);
+
+    const handleOutSystemsRedirection = () => {
```

Comment:
Previously the logix was in component signup-button.jsx . In order to extract it out , I created this hook .

Reply:
can we move this logic to shared as the same logic exists in signup-button.tsx?

Reply:
I moved it to hooks packages deriv/hooks . As it was making more sense there